docs/ref/settings.txt

```diff
@@ -1314,6 +1314,22 @@ Default primary key field type to use for models that don't have a field with
     Allowing automatic migrations for the primary key of existing auto-created
     through tables :ticket:`may be implemented at a later date <32674>`.
 
+.. warning:: Changing ``DEFAULT_AUTO_FIELD`` and third-party apps
+
+    Be aware that changing ``DEFAULT_AUTO_FIELD`` will also generate migrations
+    for third-party apps, which may later conflict with migrations shipped by
+    those apps’ maintainers. Ensure you properly review any such migrations
+    before applying them.
+
+    To work around this, you can specify the type of the auto-created primary
+    key field for each app by setting the app’s
+    :attr:`AppConfig.default_auto_field <django.apps.AppConfig.default_auto_field>`
```

Comment:
Setting the app isn't clear for me, it sounds like patching the source files. I take it you meant subclassing the AppConfig and providing that AppConfig instance to `INSTALLED_APPS` instead of a string?